Calling VideoCapture::getBackendName() causes crash.

##### System information (version)
OpenCV version: `4.5.1` or `4.2.0`
Platform: `OS X 11.1`
Compiler: `Apple clang version 12.0.0 (clang-1200.0.32.29) x86_64-apple-darwin20.3.0`

##### Detailed description

Opened `VideoCapture` fails on assertion when calling `getBackendName()`:
```
libc++abi.dylib: terminating with uncaught exception of type cv::Exception: OpenCV(4.5.1) /Users/user/app/build.release/opencv/src/opencv-pe/modules/videoio/src/cap.cpp:299: error: (-215:Assertion failed) api != 0 in function 'getBackendName'

Error: signal 6:
```

##### Steps to reproduce
1. Create and open instance of `VideoCapture` (e.g. using device `0`).
2. Call `getBackendName()` method.

--- a/modules/videoio/src/cap_avfoundation.mm
+++ b/modules/videoio/src/cap_avfoundation.mm
@@ -96,7 +96,7 @@ class CvCaptureCAM : public CvCapture {
         virtual double getProperty(int property_id) const;
         virtual bool setProperty(int property_id, double value);
         virtual int didStart();
-
+        int getCaptureDomain() /*const*/ CV_OVERRIDE { return cv::CAP_AVFOUNDATION; }
     private:
         AVCaptureSession            *mCaptureSession;
         AVCaptureDeviceInput        *mCaptureDeviceInput;
@@ -137,6 +137,7 @@ public:
     virtual double getProperty(int property_id) const;
     virtual bool setProperty(int property_id, double value);
     virtual int didStart();
+    int getCaptureDomain() /*const*/ CV_OVERRIDE { return cv::CAP_AVFOUNDATION; }
 private:
     AVAsset                  *mAsset;
     AVAssetTrack             *mAssetTrack;

--- a/modules/videoio/src/cap_avfoundation_mac.mm
+++ b/modules/videoio/src/cap_avfoundation_mac.mm
@@ -99,7 +99,7 @@ public:
     virtual double getProperty(int property_id) const;
     virtual bool setProperty(int property_id, double value);
     virtual int didStart();
-
+    int getCaptureDomain() /*const*/ CV_OVERRIDE { return cv::CAP_AVFOUNDATION; }
 
 private:
     AVCaptureSession            *mCaptureSession;
@@ -141,7 +141,7 @@ public:
     virtual double getProperty(int property_id) const;
     virtual bool setProperty(int property_id, double value);
     virtual int didStart();
-
+    int getCaptureDomain() /*const*/ CV_OVERRIDE { return cv::CAP_AVFOUNDATION; }
 
 private:
     AVAsset                  *mAsset;

